test/grpc_test.rb

```diff
@@ -85,8 +85,9 @@ def test_timeout_opens_the_circuit
   def test_instrumentation
     notified = false
     subscriber = Semian.subscribe do |event, resource, scope, adapter|
+      return if event != :success
```

Comment:
`return` in a block applies to the function it's in, which in this case would be `test_instrumentation`. If you just want to not have the block executed for the notification but the rest of the function to continue, `next` is what you're looking for.

Similar for some other tests.

Reply:
Done.